pinot-core/src/test/java/org/apache/pinot/queries/ExplainPlanQueriesTest.java

```diff
@@ -0,0 +1,510 @@
+    ResultTable actual = response.getResultTable();
+
+    System.out.println(resultTableToString("Expected", expected));
+    System.out.println(resultTableToString("Actual", actual));
+
+    Assert.assertEquals(actual.getDataSchema(), expected.getDataSchema());
+    Assert.assertEquals(actual.getRows().size(), expected.getRows().size());
+
+    List<Object[]> expectedRows = expected.getRows();
+    List<Object[]> actualRows = actual.getRows();
+    for (int i = 0; i < expectedRows.size(); i++) {
+      Object[] expectedRow = expectedRows.get(i);
+      Object[] actualRow = actualRows.get(i);
+      Assert.assertEquals(actualRow.length, expectedRow.length);
+
+      for (int j = 0; j < actualRow.length; j++) {
+        Assert.assertEquals(actualRow[j], expectedRow[j]);
+      }
+    }
+  }
+
+  /** This function is used to generated expected results while creating test cases; otherwise, not used. */
+  private static String resultTableToString(String tag, ResultTable table) {
+    System.out.println("\n" + tag + ":");
+    List<Object[]> rows = table.getRows();
+
+    StringBuffer buffer = new StringBuffer();
+    for (Object[] row : rows) {
+      buffer.append("result.add(new Object[]{\"" + row[0] + "\", " + row[1] + ", " + row[2] + "});\n");
+    }
+
+    return buffer.toString();
+  }
+
+  @Test
+  public void testSelect() {
+    String query1 = "EXPLAIN PLAN FOR SELECT * FROM testTable";
+    List<Object[]> result1 = new ArrayList<>();
+    result1.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result1.add(new Object[]{"COMBINE_SELECT", 1, 0});
+    result1.add(
+        new Object[]{"SELECT(selectList:invertedIndexCol1, invertedIndexCol2, invertedIndexCol3, jsonIndexCol1, "
+            + "noIndexCol1, noIndexCol2, noIndexCol3, rangeIndexCol1, rangeIndexCol2, rangeIndexCol3, "
+            + "sortedIndexCol1, textIndexCol1)", 2, 1});
+    result1.add(
+        new Object[]{"TRANSFORM_PASSTHROUGH(transformFuncs:invertedIndexCol1, invertedIndexCol2, invertedIndexCol3, "
+            + "jsonIndexCol1, noIndexCol1, noIndexCol2, noIndexCol3, rangeIndexCol1, rangeIndexCol2, rangeIndexCol3, "
+            + "sortedIndexCol1, textIndexCol1)", 3, 2});
+    result1.add(
+        new Object[]{"PROJECT(sortedIndexCol1, noIndexCol3, rangeIndexCol1, rangeIndexCol2, jsonIndexCol1, "
+            + "invertedIndexCol1, noIndexCol2, invertedIndexCol2, noIndexCol1, invertedIndexCol3, rangeIndexCol3, "
+            + "textIndexCol1)", 4, 3});
+    result1.add(new Object[]{"FILTER_MATCH_ENTIRE_SEGMENT(docs:1)", 5, 4});
+    check(query1, new ResultTable(DATA_SCHEMA, result1));
+
+    String query2 = "EXPLAIN PLAN FOR SELECT 'mickey' FROM testTable";
+    List<Object[]> result2 = new ArrayList<>();
+    result2.add(new Object[]{"BROKER_REDUCE(limit:10)", 0, -1});
+    result2.add(new Object[]{"COMBINE_SELECT", 1, 0});
+    result2.add(new Object[]{"SELECT(selectList:'mickey')", 2, 1});
+    result2.add(new Object[]{"PROJECT(ALL)", 3, 2});
```

Comment:
This is confusing. Why use `PROJECT (ALL)` when we are only selecting / projecting a single column. In fact, this is counter-intuitive. Previous query uses `SELECT *` so `PROJECT(ALL)` makes sense there but we don't use `PROJECT (ALL)`

Reply:
It should probably be `PROJECT()` since the statement is projecting a literal value, but not any columns. I am not sure if there is a lot of value in listing all the columns of the PROJECT in the explain plan output specially since we are querying only one table? What do you think about just putting a count of columns being projected?

Reply:
I think it should have the name of columns for clarity.